Frontend/js/raytracing.js

```diff
@@ -34,11 +32,21 @@ function addShape() {
 		return;
 			
 	}
+	
+	//translate to the center of the canvas
+	y = -y + c.height/2;
+	x = x + c.width/2;
+		
+	//set values depending on z(depth)
+	size = size * 50;
+	var sizeNew = (size * 15)/z;
+	var xNew = ((x - c.width/2)  * 10 / z) + c.width/2 ;

```

Comment:
I think we need to use more descriptive variables. 'xNew' is not a great name. I think if we use something like 'x_coord' or something, It'll be more descriptive and will help the other developers understand the code better. 'xNew' is new today but will be old in a few days and we will have to change the names again then. So, if it is general, then it'll be great.

Reply:
Same with 'yNew' and 'sizeNew'.

Reply:
Also, please add the validations Farhad added for the coordinates.